momento-sdk/src/main/java/momento/sdk/retry/FixedCountRetryStrategy.java

```diff
@@ -0,0 +1,42 @@
+package momento.sdk.retry;
+
+import com.google.common.base.Preconditions;
+import java.util.Optional;
+
+/**
+ * A retry strategy that retries a failed gRPC call a fixed number of times, with no delay between
+ * retry attempts.
+ *
+ * <p>The `FixedCountRetryStrategy` implements the {@link RetryStrategy} interface and provides a
+ * fixed number of retry attempts without any delay between retries. It allows clients to specify
+ * the maximum number of retry attempts.
+ *
+ * <p>When a gRPC call encounters an error, the `FixedCountRetryStrategy` retries the call a total
+ * of maxAttempts times. The strategy does not introduce any delay between retries, effectively
+ * retrying the call immediately for the fixed number of attempts specified.
+ */
+public class FixedCountRetryStrategy implements RetryStrategy {
+  private final int maxAttempts;
+
+  /**
+   * Constructs a `FixedCountRetryStrategy` with the provided maximum number of retry attempts.
+   *
+   * @param maxAttempts The maximum number of retry attempts. After reaching this limit, no more
+   *     retries will be performed, and the strategy will return an empty optional.
+   * @throws IllegalArgumentException if maxAttempts is not greater than 0.
+   */
+  public FixedCountRetryStrategy(final int maxAttempts) {
+    Preconditions.checkArgument(
+        maxAttempts > 0, "Total number " + "of retry attempts should be greater than 0");
+    this.maxAttempts = maxAttempts;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Optional<Long> getDelay(final int currentAttempt) {
```

Comment:
you probably saw that in the other sdks we call this `determineWhenToRetry` or something.

Your name is totally reasonable and in a green field I'd definitely have been open to discussion between those two names.

But given that some of our other SDKs are more mature and widely used than this one I think there's at least a small argument to be made that consistency would be valuable.

I don't feel super strongly about it, it would only ever be relevant to a polyglot developer, and on top of that someone who needed an advanced/custom retry strategy, and even then it's probably not hard for someone to figure it out.

So I'm happy to let you make the choice whether the consistency is worthwhile here.

Reply:
I dont have a strong opinion either :) I will make it consistent as that sounds something we should do!